lib/pure/results.nim

```diff
@@ -0,0 +1,124 @@
+import typetraits, strutils
+
+type
+  Result*[T] = object
+    ## Either a value or an error.
+    case isSuccess*: bool:
+    of true:
+      value*: T
+    of false:
+      error*: ref Exception
+      backtrace*: BacktraceFrame
```

Comment:
I don't know a way to implement async backtraces exceptions without using linked list. You can't just append to string, because one error Result may be 'forked' and used in multiple places.